internal/providers/terraform/google/google.go

```diff
@@ -37,15 +37,30 @@ func GetResourceRegion(resourceType string, v gjson.Result) string {
 }
 
 func ParseTags(r *schema.ResourceData) *map[string]string {
-	_, supportsTags := provider_schemas.GoogleLabelsSupport[r.Type]
+	_, supportsLabels := provider_schemas.GoogleLabelsSupport[r.Type]
 	rLabels := r.Get("labels").Map()
-	if !supportsTags && len(rLabels) == 0 {
+
+	_, supportsUserLabels := provider_schemas.GoogleUserLabelsSupport[r.Type]
+	rUserLabels := r.Get("user_labels").Map()
+
+	_, supportsSettingsUserLabels := provider_schemas.GoogleSettingsUserLabelsSupport[r.Type]
+	rSettingsUserLabels := r.Get("settings.0.user_labels").Map()
+
+	if !supportsLabels && len(rLabels) == 0 &&
+		!supportsUserLabels && len(rUserLabels) == 0 &&
+		!supportsSettingsUserLabels && len(rSettingsUserLabels) == 0 {
 		return nil
 	}
```

Comment:
I'm a little unsure why we are checking both if the labels are supported and the length of the labels. Surely just checking if they are supported is sufficient? If it is not, I'm unsure why we actually use the supported schemas?

Reply:
This is just a small safety factor since we don't try to match the provider version used by the tf being parsed.  I don't think it hurts anything.